### DISCUSSION TITLE
Raspberry Pi's are impossible to find. What do I do if I want to run Klipper on my machine?

### DISCUSSION THREAD

**Discussion by bassamanator** [bassamanator/Sovol-SV06-firmware#8] [Announcements]
#### Locate Raspberry Pis via https://rpilocator.com/
You can run Klipper on a very low powered computer such as the Raspberry Pi Zero. Some time ago, I paid $18 (plus shipping and taxes) for my [Raspberry Pi Zero 2 W](https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-zero-2-w). The device was very appropriately priced. No one should pay more than $25 for such a low powered device, unless you specifically need a very small, low powered device. It doesn't make any sense otherwise. Right now, to get a similarly powered Raspberry Pi, or alternative such as the OrangePi, you'd have to pay roughly $50 + shipping. I think that's absurd.

So what do you do if you want to run Klipper on your 3D printer? Use any old laptop or computer, install pretty much any distribution of Linux on it, and you're good to go. If you want to keep things **very simple** because all the tutorials that you see on Youtube are using Klipper on a Raspberry Pi, well just install [Raspberry Pi Desktop for PC and Mac](https://www.raspberrypi.com/software/raspberry-pi-desktop/) on your old laptop or PC (or Mac apparently) and follow the tutorial exactly as-is.

What if you don't have an old computer to use, what then? Well, then I recommend purchasing a cheap, used computer. Any computer that has 1 USB port and someway to connect to a network will be fine.

Having said that, I like the mini PCs by Lenovo and HP. At the time of writing, this used [Lenovo ThinkCentre M73 Tiny Form Factor Business Desktop Computer](https://www.amazon.ca/Lenovo-ThinkCentre-M73-Tiny-Refurbished/dp/B073SJQW4B/ref=sr_1_1?crid=PAI0FWZV5GON&keywords=lenovo+thinkcentre+m73&qid=1678916449&sprefix=lenovo+thinkcentre+m73%2Caps%2C115&sr=8-1) is $89 on Amazon, I'm sure you can find them cheaper on eBay or Facebook marketplace. On [this link](https://www.cpubenchmark.net/compare/2189vs4297/Intel-Pentium-G3220T-vs-BCM2711) you get an idea of the difference between the compute power of the Lenovo and the Raspberry Pi. These are business grade computers btw, they are not flimsy machines. To me, there is no question as to what you should buy *at this point in time*.

Hope this rant helps you decide what to buy.

PS I have no gripes with the Raspberry Pi. I own 4 and am attached to all of them (except maybe the Rpi3, but that's another story). They are excellent machines when purchased at MSRP.
- RPi 4-8GB
- RPi 4-2GB
- RPi 3 of some kind
- RPi Zero 2 W

\* *All prices mentioned are in Canadian rubles (CAD)*

**Comment by fkam18** [comment]
Haven't been playing around with RPi for quite a while and just checked like you said in UK the RPi 4 is over £1xx... Kidding  LoL
Like you said, go get a used i3 mini pc running Debian might be better, that on ebay could be like £30 or less, and perhaps using some 20watt.

**Comment by MihkelBaranov** [comment]
Orange Pi's is also a relatively cheap option, I'm running my stuff on an OrangePI Zero 2 https://www.aliexpress.com/item/1005004018392265.html?spm=a2g0o.order_list.order_list_main.27.506a18021ty7Uu

**Comment by asusdk** [comment]
If you have an old Laptop or Computer, you just can install Ubuntu Server, and use kiauh to install klipper. Works just fine (besides, you can't use an adxl (at least I found no way ^^)). I for example use a Laptop (thinkpad x202 tablet) and run Proxmox, and in there, I run 2 VM´s all with Ubuntu server and each of these controls one printer at the same time, while also i have running 3 Containers running pihole, unbound and lancache running on the same machine at the same time without any problems.

**Reply by Altduckthing** [reply]
There's is a way to adxl on laptop. Just buy one of the many usb input shaper and follow the tutorials. Or you could make one with a pi pico.

**Comment by Maxys-original** [comment]
Im on Orange Pi 3LTS

**Comment by dk1aj** [comment]
I use a thin client from HP
e.g. 
<https://www.ebay.com/sch/i.html?_from=R40&_trksid=p2334524.m570.l1313&_nkw=hp+t630&_sacat=0&LH_TitleDesc=0&_odkw=hp+t620&_osacat=0>

**Reply by bassamanator** [reply]
Love the price of these!
